tools/ui-components/src/close-button/close-button.tsx

```diff
@@ -15,18 +15,41 @@ export function CloseButton({
   onClick
 }: CloseButtonProps): JSX.Element {
   const classes = [
-    'text-xl font-bold leading-none appearance-none opacity-20',
-    'hover:opacity-50 focus:opacity-50',
+    // Remove browser's default styles
+    'bg-transparent',
+    'border-none',
+    // Text styles
+    'text-lg',
+    'font-bold',
+    'text-foreground-primary',
+    // Active state
+    'active:before:w-full',
+    'active:before:h-full',
+    'active:before:absolute',
+    'active:before:inset-0',
+    'active:before:border-3',
+    'active:before:border-transparent',
+    'active:before:bg-gray-900',
+    'active:before:opacity-20',
```

Comment:
True, it's controlled by the consumers, but I think it's safe to make that assumption, there is a general button component for when they need a button that doesn't close something, and we allow className, so they can add these classes for their unique cases. 

I am not experienced, so this may be common more than I think, so I am leaving this to your judgment. if you think this worth keeping, add the pseudo-element, because the styles aren't affecting it, because it doesn't have content, I think

Reply:
> we asked them to build a separate component themselves if we don't think the feature should be part of the library 

I do like that so much, there is a `CloseButton` in the `/learn`, and I won't be using this component in the `Modal`, because I don't need it. we build the component in `/learn`.

This way, we won't have to worry about maintaining the library with `react-i18next` in it, and if freeCodeCamp have to move to a different solution for i18n, we won't have to worry about it being here in the library.

We can merge this as is, and when the `Button` PR is merged, and the decoupling of `Link` and `Button` happen, we can create a PR that delete this component and move it to learn, this way the `active` that are used for links aren't used in the buttons, and we won't have to worry about maintaining translation library feature here.

---

<details>
<summary>Context of what I know about the future of the library</summary>

This is all the context that I have, I will focus the current implementation of the UI components library only on /learn, the future implementation will start to account for how to handle unique cases. 

Why is that? because I don't know which next milestone will be after we move away from bootstrap.

if we focus on `/news`, then we have to decide if we are moving `/news` to react or making web-components, aka `<fcc-button></fcc-button>`, and this library will act as wrapper for `/learn` usage.

if we focus on `/publish` and `/classroom`, we will have to account for these situations and if they do happen in there, but they would already have made their own features, and it will be the other way around, do we need the feature in `/classroom` and `/publish` in `/learn`? I personally think we won't.
</details>

Reply:
I have linked your point in ui-component issue, so it doesn't get lost easily. I am resolving this, because further discussion on how to apply this, may block this PR.